### ISSUE TITLE
`LOAD FRAGMENT` is not allowed but rgbasm(5) says it is

### ISSUE THREAD

**Issue by Rangi42** [gbdev/rgbds#1535]
Either the source or the docs needs correcting.

> `LOAD` blocks can use the `UNION` or `FRAGMENT` modifiers, as described below.

```asm
section "rom", rom0
db 1, 2, 3
load fragment "ram", wram0
db 4, 5, 6
endl
```

```
error: foo.asm(3):
    `LOAD FRAGMENT` is not allowed
error: foo.asm(5):
    Found `ENDL` outside of a `LOAD` block
```

**Comment by JL2210** [comment]
I'm pretty sure `LOAD` can't use either `UNION` or `FRAGMENT`, seeing as unions have to be in RAM. If this was implemented I'd like a way to distinguish between the start of a LOAD block in RAM and the start of a LOAD block in ROM. (Maybe `STARTOF(LOAD "section name")` vs `STARTOF("section name")`?)

**Comment by Rangi42** [comment]
There's no such thing as a `LOAD` block in RAM. Section names are unique; `STARTOF("foo")` refers to the one section "foo", whatever type it happens to be. A `LOAD block declares a RAM section "within" a ROM section: the data declarations keep on applying to the current ROM section, but the labels within the `LOAD` block go within the RAM section.

**Comment by ISSOtm** [comment]
Where the block starts in ROM can be obtained by putting a label right before the `LOAD` directive, fwiw.

**Comment by JL2210** [comment]
ah, I see. So load blocks need to have a section in ROM as well as a section in RAM, even if they both are supposed to "reference" the same place

**Comment by JL2210** [comment]
Would it make sense to make load fragments require that their section be fragmented as well?

**Comment by aaaaaa123456789** [comment]
> Would it make sense to make load fragments require that their section be fragmented as well?

```
SECTION "Test", ROMX

  db 1
  LOAD FRAGMENT "Load test", WRAMX
  db 2
  ENDL
  db 3
  LOAD FRAGMENT "Load test", WRAMX
  db 4
  ENDL
```

Does this make sense? Sure it does: the semantics are pretty obvious. No reason to impose a requirement on the section.

**Comment by Rangi42** [comment]
Yeah, and we already support `LOAD UNION` (I've used it in PC), so if there's no technical hurdle to `LOAD FRAGMENT` *and* the docs already claim it works, then it should.

**Comment by Rangi42** [comment]
Oh yeah! We *used* to support `LOAD FRAGMENT` but removed it in 1d012682497e5060df257a12c2a3e3388d43e4ab (and just didn't update the docs). #869 has more explanation:

> `LOAD FRAGMENT` needs to be removed anyway, as its semantics are buggy.
All sections in RGBDS are assumed to be contiguous; this is the basis for a *lot* of the implied behavior. `LOAD FRAGMENT` breaks that assumption.
>
> `LOAD` causes a "RAM" section to mirror (size-wise and label-wise) a segment of a "data" section. This makes sense. `LOAD UNION` merely allows the RAM block of data to share space with others, like the well-known "overlay" feature present in a lot of systems.
>
> `LOAD FRAGMENT`, however, enables breaking the RAM section's *actual* contents across several, non-contiguous ROM sections. This breaks the implicit assumption that there is a 1:1 mapping between the ROM and "RAM" sections.

I'm now unsure if we can feasibly add it back. @ISSOtm what do you think?

**Comment by JL2210** [comment]
My use case doesn't require the ROM section that the `LOAD FRAGMENT` is in to contain anything except `LOAD FRAGMENT`s. I would expect that, at least, to have sane semantics. I'd like it to be added back either way; if the section containing `LOAD FRAGMENT`s must only comprise of `LOAD FRAGMENT`s, that's fine by me.

**Comment by JL2210** [comment]
Alternatively a less strict requirement could be that all (combined) `LOAD FRAGMENT`s be both in the same section (or fragment) and contiguous with one another.
